### ISSUE TITLE
Wrong suggestion for an edge case of Alternate Value

### ISSUE THREAD

**Issue by kvelicka** [koalaman/shellcheck#799]
#### For bugs
- My shellcheck version (`shellcheck --version` or "online"): `0.4.5` and online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
countArgs () { echo "$#"; }
brr=( '' )
countArgs ${brr[@]+"${brr[@]}"} # prints 1 (correct answer) but shellcheck complains
countArgs "${brr[@]+"${brr[@]}"}" # prints 0 (incorrect) but shellcheck doesn't complain
```

#### Here's what shellcheck currently says:
```
Line 3:
countArgs ${brr[@]+"${brr[@]}"} # prints 1 (correct answer) but shellcheck complains
          ^-- SC2068: Double quote array expansions to avoid re-splitting elements.
```

#### Here's what I wanted or expected to see:
Nothing I guess? This could even be a `bash` bug, or a very obscure edge case. The expressions used in the snippet above are taken from [this](http://stackoverflow.com/questions/7577052/bash-empty-array-expansion-with-set-u/7577209#7577209) StackOverflow answer.

**Comment by brother** [comment]
Not sure what bash version you use but I get 1 as result in both the cases. Just a heads up. No complaints on the other parts =)

```shell
brother ~$ cat /tmp/799.sh 
#!/bin/bash
countArgs () { echo "$#"; }
brr=( '' )
countArgs ${brr[@]+"${brr[@]}"} # prints 1 (correct answer) but shellcheck complains
countArgs "${brr[@]+"${brr[@]}"}" # prints 0 (incorrect) but shellcheck doesn't complain
brother ~$ /tmp/799.sh 
1
1
brother ~$ shellcheck /tmp/799.sh

In /tmp/799.sh line 4:
countArgs ${brr[@]+"${brr[@]}"} # prints 1 (correct answer) but shellcheck complains
          ^-- SC2068: Double quote array expansions to avoid re-splitting elements.

```

**Comment by kvelicka** [comment]
Hmm, that's very weird! What version of bash are you using? I tried doing the same thing as you did in a fresh ubuntu:16.04 container:

```bash
root@dfb5f6d921ac:/# cat /tmp/799.sh
#!/bin/bash
countArgs () { echo "$#"; }
brr=( '' )
countArgs ${brr[@]+"${brr[@]}"} # prints 1 (correct answer) but shellcheck complains
countArgs "${brr[@]+"${brr[@]}"}" # prints 0 (incorrect) but shellcheck doesn't complain

root@dfb5f6d921ac:/# /tmp/799.sh
1
0
root@dfb5f6d921ac:/# bash --version
GNU bash, version 4.3.46(1)-release (x86_64-pc-linux-gnu)
<snip>
```

The machine on which I discover this [potential] bug is a ubuntu 16.04 machine as well, with the same version of bash.

**Comment by brother** [comment]
GNU bash, version 4.4.5(1)-release (x86_64-pc-linux-gnu)

**Comment by kvelicka** [comment]
I see. Well, in that case this issue might be considered invalid, since it looks more like a `bash` bug. Kind of annoying that it's in the version that ships with a LTS version of Ubuntu though!

**Comment by kvelicka** [comment]
Yup, this most likely is a bash bug.
